### PR TITLE
Feature/disable wal flag

### PR DESCRIPTION
**Pull request by asdacap**
- The WAL files in rocksdb is its durability mechanism which allow it to recover writes in crashes where the memtable was not flushed to the disk. 
- In case of snapsync or full pruning, if it crash, it will have to start over meaning the WAL is unnecessary, so disabling it saves some writes. The amount saved should be roughly the size of the DB, which is around 160GB.
- For full pruning (32GB mem budget), total writes goes down from 1.7TB to 1.5TB. Total time remains at 40minute, probably because SSD warmed up or some other limit.
- For snap sync (HeavyWrite tune), total writes goes down from 3.28TB to 2.93TB. Took similar amount of time (1 hour vs 50 minute). 
- I guess instead of `DisableWAL` a better more universal term for flag would be `NoDurable`, `NonDurable`, `DurabilityNotRequired` or something, but can't find a good one. Got idea?

## Changes

- Add `DisableWAL` write flag. Integrate with RocksDB, SnapSync and FullPruning.
- Slight change with snap sync, code save where it uses write batch instead.
- Refactor rocksdb's smoke test to use same test for column and non-column db.
- Add an explicit flush when full pruning complete.

#### What types of changes does your code introduce?

- [X] Optimization
- [X] Refactoring

#### Requires testing

- [X] Yes
- [ ] No

#### If yes, did you write tests?

- [X] Yes
- [ ] No

#### Notes on testing

_Optional. Remove if not applicable._

## Remarks

_Optional. Remove if not applicable._

### CODE REVIEW

File: src/Nethermind/Nethermind.Blockchain/FullPruning/FullPruner.cs [hunk 1]
```diff
@@ -182,6 +182,8 @@ protected virtual void RunPruning(IPruningContext pruning, Keccak statRoot)
                     writeFlags = WriteFlags.None;
                 }
 
+                writeFlags |= WriteFlags.DisableWAL;
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Blockchain/FullPruning/FullPruner.cs:185]: Why not start with it and then set `LowPriority` based on `_pruningConfig.FullPruningDisableLowPriorityWrites` ?

File: src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs [hunk 2]
```diff
@@ -35,9 +35,10 @@ public class DbOnTheRocks : IDbWithSpan, ITunableDb
 
     private IntPtr? _rateLimiter;
     internal WriteOptions? WriteOptions { get; private set; }
-    internal WriteOptions? LowPriorityWriteOptions { get; private set; }
-
-    internal ReadOptions? _readAheadReadOptions = null;
+    private WriteOptions? _noWalWrite;
+    private WriteOptions? _lowPriorityAndNoWalWrite;
+    private WriteOptions? _lowPriorityWriteOptions;
+    private ReadOptions? _readAheadReadOptions = null;
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs:41]: Can/should they be static?

File: src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs [hunk 3]
```diff
@@ -499,9 +514,19 @@ public void Set(ReadOnlySpan<byte> key, byte[]? value, WriteFlags flags = WriteF
 
     public WriteOptions? WriteFlagsToWriteOptions(WriteFlags flags)
     {
-        if (flags == WriteFlags.LowPriority)
+        if ((flags & WriteFlags.LowPriority) != 0 && (flags & WriteFlags.DisableWAL) != 0)
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs:517]: Either save flags to `bool` and reuse them or check for both flags in one go.

File: src/Nethermind/Nethermind.Blockchain/FullPruning/FullPruner.cs (hunk 1 above)
Comment (asdacap) [reply src/Nethermind/Nethermind.Blockchain/FullPruning/FullPruner.cs:185]: That would do too.

File: src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs (hunk 2 above)
Comment (asdacap) [reply src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs:41]: Maybe.

File: src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs (hunk 3 above)
Comment (asdacap) [reply src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs:517]: Ok

File: src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs (hunk 2 above)
Comment (asdacap) [reply src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs:41]: Ah, `WriteAheadLogSync` is from config.

**Pull request merged**